src/main/java/data_access/CachingRateMyProfDataAccess.java

```diff
@@ -0,0 +1,38 @@
+package data_access;
+
+import entity.Professor;
+import use_case.ratemyprof.RateMyProfDataAccessInterface;
+
+import java.util.*;
+
+/**
+ * This class caches RateMyProf calls to improve performance and lessen the load on underlying data source.
+ */
+public class CachingRateMyProfDataAccess implements RateMyProfDataAccessInterface {
+    private final RateMyProfDataAccessObject innerFetcher;
```

Comment:
```suggestion
    private final RateMyProfDataAccessInterface innerFetcher;
```
innerFetcher is currently typed as RateMyProfDataAccessObject, but the class already depends on RateMyProfDataAccessInterface. For flexibility and testability, it would be better to type this field \ as RateMyProfDataAccessInterface.